cmd/nsm/main.go

```diff
@@ -20,10 +20,9 @@ func main() {
 		syscall.SIGINT,
 		syscall.SIGTERM,
 		syscall.SIGQUIT)
-	select {
-	case s := <-sigChan:
+	go func() {
```

Comment:
Is this still producing a race condition?

Reply:
What are we trying to accomplish here?

Reply:
Try using this to block main from exiting.

https://golang.org/pkg/sync/#WaitGroup

Reply:
OK, I've got an update which adds WaitGroup support, and when coupled with #29, this now works as expected! New commit pushing in 30 seconds!